packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -452,6 +492,53 @@ export class StateMachine extends StateMachineBase {
       }
     }
 
+    if (props?.kmsKey) {
+      props.kmsKey.addToResourcePolicy(new iam.PolicyStatement({
+        resources: ['*'],
+        actions: ['kms:Decrypt', 'kms:GenerateDataKey', 'kms:DescribeKey'],
+        principals: [new iam.ServicePrincipal('states.amazonaws.com')],
+        conditions: {
+          StringEquals: {
+            'aws:SourceAccount': this.stack.account,
+            'aws:SourceArn': Stack.of(this).formatArn({
+              service: 'states',
+              resource: 'stateMachine',
+              resourceName: this.physicalName,
+            }),
+            'kms:EncryptionContext:aws:states:stateMachineArn': Stack.of(this).formatArn({
+              service: 'states',
+              resource: 'stateMachine',
+              resourceName: this.physicalName,
+            }),
+          },
+        },
+      }));
+
+      if (props?.enableEncryptedLogging && props?.logs) {
```

Comment:
The docs are very unclear, but after testing this myself, I realized this statement MUST BE in the key policy if encryption is enabled otherwise there will be no logs at all for the state machine. This should be added to the policy whenever logging and encryption are both enabled. Remove the `enableEncryptedLogging` prop.

```json
{
      "Sid": "Enable log service delivery for integrations",
      "Effect": "Allow",
      "Principal": {
        "Service": "delivery.logs.amazonaws.com"
      },
      "Action": "kms:Decrypt*",
      "Resource": "*"
}
```

Encrypting the entire log group (what's being added above on line 518) is unrelated to Step Functions, so it should be removed from the state machine construct.

Reply:
I see, will remove the logic to encryt the entire log group and the `enableEncryptedLogging` prop